File: matlab/+copilot/SdkProtocolVersion.m
classdef SdkProtocolVersion
    % SdkProtocolVersion  Protocol version negotiated with the Copilot CLI.
    %
    %   The version must match the value in sdk-protocol-version.json at the
    %   repository root. Bump this constant when the upstream protocol changes.

    properties (Constant)
        VERSION = 3
    end
end

File: visualbasic/src/SdkProtocolVersion.vb
' Copyright (c) Microsoft Corporation. All rights reserved.

Namespace GitHub.Copilot.SDK

    ''' <summary>
    ''' Provides the SDK protocol version constant.
    ''' This must match the version expected by the copilot-agent-runtime server.
    ''' </summary>
    Public Module SdkProtocolVersion

        ''' <summary>
        ''' The SDK protocol version string.
        ''' </summary>
        Public Const Version As String = "2"

        ''' <summary>
        ''' Gets the protocol version as an integer.
        ''' </summary>
        Public Function GetVersion() As Integer
            Return Integer.Parse(Version)
        End Function

    End Module

End Namespace
